### ISSUE TITLE
IO logging doesn't work (failed to read fd 12)

### ISSUE THREAD

**Issue by soratobukuroneko** [sudo-project/sudo#151]
I tried to setup sudo IO logging on CentOS 7-2009. But it's not working. The log files (stderr, stdout, stdin) are created but they are all empty.

I use the following sudoers directives:
```
Defaults log_input, log_output, requiretty
Defaults iolog_dir = /var/log/sudo/io/%{user} # same issue if not set
Defaults iolog_file = %{runas_user}-%{command}-XXXXXX # same issue if not set
```

In sudo debug log I see "error reading fd 12: Input/output error". Here the full log:
[sudo.log](https://github.com/sudo-project/sudo/files/8806102/sudo.log)

**Comment by millert** [comment]
That I/O error usually just indicates that the pseudo-terminal has been destroyed and is not usually an error.  Since the I/O logs are compressed there is buffering involved that may result in empty I/O log files until the command completes.  From your debug log there should be 7 bytes of (uncompressed) data in the ttyout file.  Is the timing file also empty?

What version of sudo are you running?  There was a bug fixed in sudo 1.8.17 where some output buffers were not being flushed on exit.  The current sudo package for CentOS 7 appears to be sudo-1.8.23-10.el7_9.2.  I did a quick test on a CentOS 7 VM and I/O logging works as expected.

**Comment by soratobukuroneko** [comment]
Sorry I missunderstood how it worked. I don't have anything in the std* files but I do in the ttyout and sudoreplays works.
